common/src/main/java/io/netty/util/internal/CleanerJava6.java

```diff
@@ -72,31 +62,21 @@
         }
         CLEANER_FIELD_OFFSET = fieldOffset;
         CLEAN_METHOD = clean;
-        CLEANER_IS_RUNNABLE = cleanerIsRunnable;
-
-        // free buffer if possible
-        freeDirectBuffer(direct);
     }
 
-    static void freeDirectBuffer(ByteBuffer buffer) {
+    @Override
+    public void freeDirectBuffer(ByteBuffer buffer) {
         if (CLEANER_FIELD_OFFSET == -1 || !buffer.isDirect()) {
             return;
         }
-        assert CLEAN_METHOD != null || CLEANER_IS_RUNNABLE:
-                "CLEANER_FIELD_OFFSET != -1 implies CLEAN_METHOD != null or CLEANER_IS_RUNNABLE == true";
+        assert CLEAN_METHOD != null;
         try {
             Object cleaner = PlatformDependent0.getObject(buffer, CLEANER_FIELD_OFFSET);
             if (cleaner != null) {
-                if (CLEANER_IS_RUNNABLE) {
-                    ((Runnable) cleaner).run();
-                } else {
-                    CLEAN_METHOD.invoke(cleaner);
-                }
+                CLEAN_METHOD.invoke(cleaner);
             }
         } catch (Throwable t) {
             // Nothing we can do here.
```

Comment:
yah if we never expect it to happen lets re-throw it. What about `PlatformDependent.throwException(t);`?